# Whole Exome Sequencing Breast Cancer Variant Calling

## Introduction
This project follows the basic steps outlined in good practices for identifying variants in a Whole Exome Sequencing sample using the Galaxy platform. The analysis consists of the following steps:  
<p align="center"> <img src="Recursos/Workflow.png" alt="Workflow"> </p>

---

## Sample Information
The analyzed sample comes from breast tissue of a 53-year-old Chinese woman. For exome capture, the TruSeq Exome Library Preparation Kit was used, and sequencing was performed using the high-throughput Illumina NextSeq 500 platform, generating paired-end reads. This dataset is available in the SRA repository under the accession number [SRR9899132](https://www.ncbi.nlm.nih.gov/sra/?term=SRR9899132).

---

# Methodology

## Quality Control Analysis

### Basic Statistics
To evaluate the quality of the generated reads, **FastQC** was used, a widely employed tool for this type of analysis. According to the basic statistics provided by FastQC, the GC content is **49%**, which deviates from the expected average of **41%** in the whole human genome. This deviation is expected and correct since the experiment is a **whole exome sequencing (WES)** capture, which enriches exons, typically having higher GC content compared to the rest of the genome.  
<p align="center"> <img src="Recursos/Basic_statistics_raw_reads.png" alt="FastQC Basic Statistics"> </p>

---

### Sequence Quality per Base
To evaluate the quality of the generated reads, the per-base sequence quality graph was generated using Sanger/Illumina 1.9 encoding. This graph is a crucial tool for identifying potential issues in read quality.  
<p align="center"> <img src="Recursos/Per_base_sequence_quality_raw_reads.png" alt="Per Base Sequence Quality Graph"> </p>

---

### Sequence Content per Base
In the **Per base sequence content** section, a **deregulation in nucleotide percentages** is detected in the final positions. This behavior is particularly noticeable for **adenine (A)**, which shows a percentage of **0%** in the last analyzed position.  
<p align="center"> <img src="Recursos/Per_base_sequence_content_raw_reads.png" alt="Graph Before Trim Galore Filtering"> </p>

---

## Read Filtering
To improve read quality, Trim Galore was used with the --three_prime_clip_R1/2 option, removing the last 5 bases (5 bp) that presented quality issues at the 3' ends. Following S. P. Pfeifer's recommendations in [_From next-generation resequencing reads to a high-quality variant data set_](https://pmc.ncbi.nlm.nih.gov/articles/PMC5234474), these reads fall under "Potential issue 1: low-quality data" and follow the dynamics described in [Figure 2b](https://pmc.ncbi.nlm.nih.gov/articles/PMC5234474/#fig2), making their removal advisable.  
<p align="center"> <img src="Recursos/Per_base_sequence_content_trim_reads.png" alt="Graph After Trim Galore Filtering"> </p>

---

## Read Mapping

The mapping process was performed using **BWA-MEM**, a widely used tool for read alignment in genomic analyses. According to the study [_Assembling and Validating Bioinformatic Pipelines for Next-Generation Sequencing Clinical Assays_](https://www.tandfonline.com/doi/full/10.2144/000114492), Burrows-Wheeler Aligner (BWA) offers more accurate alignment compared to Bowtie2, resulting in more reliable variant calls, making it the preferred choice for clinical applications.

The choice of BWA-MEM was based on its ability to handle long and complex reads with high precision, ideal for experiments like exome capture, where mapping accuracy is crucial for minimizing errors in variant calling and ensuring high-quality data.

Additionally, the reference genome **GRCh38** was used, as indicated in [_Assembling and Validating Bioinformatic Pipelines for Next-Generation Sequencing Clinical Assays_](https://www.tandfonline.com/doi/full/10.2144/000114492), which includes significant improvements compared to previous versions. These improvements reduce errors in variant detection, increasing the accuracy and reliability of results. Reads were sorted by coordinates by default in Galaxy. Below are the general alignment statistics generated by [**Samtools flagstat**](Recursos/Samtools%20flagstat.txt).

The total reads analyzed were **45,303,149**, of which **99.63%** were successfully mapped to the reference genome. Additionally, **99.01%** of the reads were properly mapped as correctly paired mates. The percentage of singletons was low, representing only **0.22%**.

---

## BAM File Processing

To ensure high-quality alignment and prepare data for reliable variant analysis, an exhaustive processing of the BAM file was performed. According to [_Assembling and Validating Bioinformatic Pipelines for Next-Generation Sequencing Clinical Assays_](https://meridian.allenpress.com/aplm/article/144/9/1118/427496/Assembling-and-Validating-Bioinformatic-Pipelines), perfectly aligned reads have an average mapping quality score (MAPQ) of **60** (1/10⁻⁶, meaning a 0.0001% probability of incorrect alignment). Conversely, a score below **30** (1/10⁻³, a 0.1% error probability) is generally considered unacceptable.

Using **Filter BAM**, a filter was applied to select reads with **MAPQ ≥30**, discarding lower-quality reads. Additionally, all unmapped or mate-unmapped reads were excluded to obtain a cleaner and higher-quality dataset.

To evaluate the impact of this filtering, **Qualimap** was used, revealing the following results:  
- Before filtering, the mean mapping quality score was **27.27**.  
<p align="center"> <img src="Recursos/MAPQ_raw_BAM_file.png" alt="MAPQ Before Filtering"> </p>

- After filtering, the mean increased significantly to **40.28**, indicating a substantial improvement in read quality.  
<p align="center"> <img src="Recursos/MAPQ_filtered_BAM_file.png" alt="MAPQ After Filtering"> </p>

This processing is essential for ensuring reliable analysis in subsequent steps, particularly in variant calling.

---

## Additional Processing Steps

According to [_From next-generation resequencing reads to a high-quality variant data set_](https://pmc.ncbi.nlm.nih.gov/articles/PMC5234474/#sec13), recommended post-mapping steps include:  
- Local realignment  
- Duplicate marking  
- Base quality recalibration  

However, in this analysis, local realignment was not performed because **FreeBayes** incorporates this adjustment internally during the variant calling process. This functionality is described in its [official documentation](https://github.com/freebayes/freebayes?tab=readme-ov-file#indels).

As for **base quality recalibration**, studies like [_Toward better understanding of artifacts in variant calling from high coverage samples_](https://pmc.ncbi.nlm.nih.gov/articles/PMC4271055/) have shown that precision improvements in variant calling after this step are marginal. Additionally, given its high computational cost, it can be considered an optional preprocessing step. For these reasons, this step was not performed in the current analysis.

---

## Duplicate Marking

The next step in BAM file processing was duplicate marking, performed using **Picard MarkDuplicates**. Before duplicate marking, the duplicate rate was **30.95%**, as shown in the previously generated graph:

<p align="center">
  <img src="Recursos/Duplicate_rate_raw_BAM_file.png" alt="Duplicate Rate Before Marking">
</p>

After running this tool, all duplicates were removed, leaving a clean BAM file with no duplicated reads. This process is essential for reducing bias and improving accuracy in subsequent analyses, such as variant identification.

---

## Variant Calling

Variant calling was performed using the **FreeBayes** tool, an efficient method for identifying genetic variants in diploid data.

---

## Variant Normalization

After variant calling, a **normalization** step was carried out using **bcftools norm**. This process is crucial for ensuring consistency in the detected variants, standardizing their representation by rearranging insertions and deletions (INDELs), and resolving potential conflicts in the VCF file.

---

## Variant Filtering

To improve the quality of data resulting from variant calling, filtering criteria were applied based on recommendations from the article [_Effective filtering strategies to improve data quality from population-based whole exome sequencing studies_](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-15-125). According to the authors, effective filtering should focus on ensuring the quality and depth of detected variants.

Thus, **bcftools view** was used with the `-i` option, applying the following criteria:
- **QUAL > 30**: Variants with a quality score above 30, reducing the likelihood of false positives.  
- **DP > 10**: A read depth greater than 10, ensuring greater confidence in the validity of the called variants.

---

## Variant Annotation

The identified variants were annotated using **SNPeff**, an efficient tool for predicting the functional impact of variants on genes and transcripts. This step is essential for associating detected variants with their possible biological effects.

---

## Variant Prioritization

To prioritize annotated variants, **VEP** (Variant Effect Predictor) was used, selecting only those variants that met the criteria indicated in [_A Practical Guide To Filtering and Prioritizing Genetic Variants_](https://www.tandfonline.com/doi/full/10.2144/000114492):

- **Exonic variants** with frameshift or nonsense effects such as:
  - `frameshift_variant`
  - `stop_gained`
  - `stop_loss`
- **Missense variants** considered biologically significant. These were classified as **deleterious** based on predictions made by tools like **SIFT**, **PolyPhen**, **LRT**, **MetaSVM**, and **Mutation Taster**.

This prioritization focused the analysis on variants most likely associated with functional or pathological effects:

- **Missense variants**: 34  
- **Frameshift variants**: 157  
- **Stop gained**: 70  
- **Stop loss**: 24

---

## Variant Discussion

Variant analysis performed with FreeBayes resulted in a total of **52,217 variants** classified into different types.

---

### Functional Classes of Variants

In functional terms, the following were detected:  
- **Missense variants**: 17,225 (43.729%), which alter the amino acid sequence and may have a significant effect.  
- **Nonsense variants**: 90 (0.228%), causing premature termination of translation.  
- **Silent variants**: 22,075 (56.042%), which do not alter the amino acid sequence.

The **missense/silent** ratio of **0.7803** suggests a balanced proportion between potentially damaging variants and those that do not affect protein functionality.

---

### Ts/Tv Ratio

The **Ts/Tv ratio** (transition/transversion ratio) calculated was **2.5334**. This value falls within the expected range for exome data, indicating high-quality mapping and variant calling. This ratio is useful as a validation metric since a value within the expected range reflects precision in SNP calls and overall quality of the analysis.

---










